File: backend/rm_service/schemas/prescription.py
"""
Prescription Schemas
"""
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class PrescriptionCreate(BaseModel):
    """Schema for creating prescription"""
    record_id: str
    drug_name: str
    drug_code: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration_days: Optional[int] = None
    notes: Optional[str] = None

class PrescriptionResponse(BaseModel):
    """Schema for prescription response"""
    prescription_id: str
    record_id: str
    drug_name: str
    drug_code: Optional[str] = None
    dosage: Optional[str] = None
    frequency: Optional[str] = None
    duration_days: Optional[int] = None
    notes: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

File: apps/lib/domain/entities/prescription.dart
/// Prescription entity
class Prescription {
  final String prescriptionId;
  final String recordId;
  final String drugName;
  final String? drugCode;
  final String? dosage;
  final String? frequency;
  final int? durationDays;
  final String? notes;
  final DateTime createdAt;
  
  const Prescription({
    required this.prescriptionId,
    required this.recordId,
    required this.drugName,
    this.drugCode,
    this.dosage,
    this.frequency,
    this.durationDays,
    this.notes,
    required this.createdAt,
  });
}

